socorro/app/fetch_transform_save_app.py

```diff
@@ -37,180 +37,71 @@
-    # are relevant.
     required_config.destination.add_option(
         'crashstorage_class',
         doc='the destination storage class',
         default='socorro.external.fs.crashstorage.FSPermanentStorage',
         from_string_converter=class_converter
     )
+
     required_config.producer_consumer = Namespace()
     required_config.producer_consumer.add_option(
         'producer_consumer_class',
         doc='the class implements a threaded producer consumer queue',
         default='socorro.lib.threaded_task_manager.ThreadedTaskManager',
         from_string_converter=class_converter
     )
-    required_config.add_option(
-        'number_of_submissions',
-        doc="the number of crashes to submit (all, forever, 1...)",
-        short_form='n',
-        default='forever'
-    )
 
     def __init__(self, config):
         super().__init__(config)
         self.waiting_func = None
-        # select the iterator type based on the "number_of_submissions" config
-        self.source_iterator = {
-            'forever': self._infinite_iterator,
-            'all': self._all_iterator,
-        }.get(config.number_of_submissions, self._limited_iterator)
-
-    # - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
-    # iterator section
-    #
-    # the following methods setup a nested hierarchy of iterators.  Each layer
-    # in the hierarchy adds a feature to the enclosed iterator.
-
-    # The  iterator at the core, the innermost, will yield a list of crash_ids
-    # from some source. A subclass may define its own source and makes it
-    # available by overriding the method "_create_iter".
-    #
-    # That innermost iterator is then wrapped by the "_basic_iterator" method.
-    # this iterator changes the form of the yielded values to the (*args,
-    # **kwargs) as required by the TaskManager.  It also gives derived classes
-    # the oportunity to run callbacks between each iteration or at the point
-    # that the innermost iterator is exhausted.
-    #
-    # The outermost layer of the iterator nesting imposes length limitations
-    # on the iterators.  Controlled by the configuration parameter
-    # "number_of_submissions" this iterator can be limit to an exact number
-    # of iterations, set to run forever, or run only until the natural
-    # exhaustion of the innermost iterator.
-    # - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
-
-    def _create_iter(self):
-        # the actual mechanism of creating the iterator to be overridden in
-        # subclasses based on what they want to iterate over.  In this default,
-        # iteration will come from the crashstorage instance defined as
-        # "source".  This works for the crashmover where the generally used
-        # source is the file system walking class.
-        return self.source.new_crashes()
+        self.source_iterator = self._infinite_iterator
```

Comment:
I can probably nix this line and rename `_infinite_iterator` to `source_iterator`.